### PR TITLE
Add limitation to b2.md

### PR DESCRIPTION
**Pull request by sedlund**
The B2 gateway does not implement reporting md5sum as an etag response from PutObject.  Add to Known Limitations.

## Description
Update docs

## Motivation and Context
Good docs are good.

## Regression
N/A

## How Has This Been Tested?
N/A

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sedlund** [comment]
With regard to; https://github.com/minio/minio/issues/7624

**Review by donatello — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Review by harshavardhana — APPROVED** [review]

**Comment by deekoder** [comment]
@eco-minio did u have any feedback?

**Review by eco-minio — CHANGES_REQUESTED** [review]
Typo on validity, looks a bit more clear as "Applications that check validity will fail."

File: docs/gateway/b2.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ mc ls myb2
 ### Known limitations
 Gateway inherits the following B2 limitations:
 
+- PutObject() does not return an md5sum of the uploaded file as an etag response.  Apps that check vailidity will fail.
```
Comment (kannappanr) [inline docs/gateway/b2.md:46]: Modified your documentation a little bit. PTAL.
```
PutObject S3 API does not return md5sum of the uploaded file as it's etag response. Applications that check validity of the uploaded object by comparing these two values will fail.
```

Comment (sedlund) [reply docs/gateway/b2.md:46]: Looks good!

Comment (harshavardhana) [reply docs/gateway/b2.md:46]: Can you update the PR with the requested change?

**Pull request merged**